src/formats/gaussformat.cpp

```diff
@@ -711,10 +725,64 @@ namespace OpenBabel
                 if (!atom)
                   break;
                 atom->SetPartialCharge(atof(vs[2].c_str()));
-
+                Mulliken->AddPartialCharge(atof(vs[2].c_str()));
                 if (!ifs.getline(buffer,BUFF_SIZE)) break;
                 tokenize(vs,buffer);
+                                    
+              } 
+            if (Mulliken->NumPartialCharges() != mol.NumAtoms()) 
+            {
+                cerr << "The number of Mulliken partial charges is not equal to the number of atoms!" << endl;
+                exit (-1);
+            }
+            Mulliken->SetAttribute("Mulliken charges");
+            mol.SetData(Mulliken);    
+          }
+        else if(strstr(buffer,"Hirshfeld charges") != NULL &&
+                strstr(buffer,"CM5 charges") != NULL)
+          {
+            /*
+              Hirshfeld and CM5 charges are printed in the
+              same block in the Gaussian log file. 
+             */
+            hasPartialCharges = true;
+            chargeModel = "Hirshfeld";
+            if (mol.HasData("Hirshfeld charges"))
+              {
+                mol.DeleteData("Hirshfeld charges");
+              }
+            if (mol.HasData("CM5 charges"))
+              {
+                mol.DeleteData("CM5 charges");
               }
+            OBPcharge *Hirshfeld = new OpenBabel::OBPcharge();
+            OBPcharge *CM5       = new OpenBabel::OBPcharge();
+            ifs.getline(buffer,BUFF_SIZE);	// column headings
+            ifs.getline(buffer,BUFF_SIZE);
+            tokenize(vs,buffer);
+            while (vs.size() >= 8 &&
+                   strstr(buffer,"Tot ") == NULL)
+              {
+                atom = mol.GetAtom(atoi(vs[0].c_str()));
+                if (!atom)
+                  break;
+                atom->SetPartialCharge(atof(vs[2].c_str()));
```

Comment:
Wait - what's the preferred partial charge? I think there either needs a way to set the preferred charge (e.g. that will be written to .mol2 and other formats) or some discussion about what charges a user will get if multiple charges are found.

Right now, the code seems to quasi-randomly pick the atom->SetPartialCharge() based on the last set that it finds in the file.

Personally, it seems like I'd prefer something like:
- ESP if available
- Hirshfeld if not ESP
- Mulliken if nothing else

But it would be helpful to have a INOPTION to pick which type of partial charge is retained.

Reply:
This is actually needed. It would be helpful to have a INOPTION to pick which type of charge. We also need to consider the situation if the user specifies a charge type that does not exist. But, I think it is better to add these features as a new patch. 
Do you agree to have it as new patch?

Reply:
That's fine.